scripts/automated_ingestion/eessi_task.py

```diff
@@ -0,0 +1,1201 @@
+                self._update_pull_request(pull_request)
+
+                return EESSITaskState.PULL_REQUEST
+
+    @log_function_entry_exit()
+    def _handle_add_pull_request(self):
+        """Handler for ADD action in PULL_REQUEST state"""
+        log_message(LoggingScope.TASK_OPS, "INFO", "Handling ADD action in PULL_REQUEST state: '%s'",
+                    self.description.get_task_file_name())
+        # Implementation for adding in PULL_REQUEST state
+        # we got here because the state of the task is PULL_REQUEST in the default branch
+        # determine branch and PR and state of PR
+        # PR is open --> just return EESSITaskState.PULL_REQUEST
+        # PR is closed & merged --> deployment is approved
+        # PR is closed & not merged --> deployment is rejected
+        feature_branch_name = self._determine_feature_branch_name()
+        # TODO: check if feature branch exists, for now ASSUME it does
+        pull_request = self._find_pr_for_branch(feature_branch_name)
+        if pull_request:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "found PR for branch '%s': '%s'", feature_branch_name, pull_request)
+            if pull_request.state == "closed":
+                if pull_request.merged:
+                    log_message(LoggingScope.TASK_OPS, "INFO",
+                                "PR '%s' is closed and merged (strange that state is PULL_REQUEST)", pull_request)
+                    # TODO: How could we ended up here? state in default branch is PULL_REQUEST but
+                    #         PR is merged, hence it should have been in the APPROVED state
+                    #    ==> for now, just return EESSITaskState.PULL_REQUEST
+                    #
+                    #       there is the possibility that the PR was updated just before the
+                    #         PR was merged
+                    #       WHY is it a problem? because a task may have been accepted that wouldn't
+                    #         have been accepted or worse shouldn't been accepted
+                    #       WHAT to do? ACCEPT/IGNORE THE ISSUE FOR NOw
+                    #       HOWEVER, the contents of the PR directory may be inconsistent with
+                    #         respect to the TaskState file and missing TaskSummary.html file
+                    #       WE could create an issue and only return EESSITaskState.APPROVED if the
+                    #         issue is closed
+                    #       WE could also defer all handling of this to the handler for the
+                    #         APPROVED state
+                    # NOPE, we have to do some handling here, at least for the tasks where their
+                    #   state file did
+                    #   --> check if we could have ended up here? If so, create an issue.
+                    #       Do we need a state ISSUE_OPENED to avoid processing the task again?
```

Comment:
I had to think here: why would the task be processed again? The next time `ingest_bundles.py` runs, the `task.determine_state()` will return the last state we wrote to the state file in the _default_ branch, which will be `PULL_REQUEST` _if_ the `_handle_add_payload_staged` handler made it to the `_update_task_states` call that sets the `next_state`). Then, indeed, this will be evaluated every time.

I'm not sure if that's a problem. Technically, even the ingested and rejected tasks are handled every time that `ingest_bundles.py` is run, no? As long as that handling is 'light enough', it's ok I guess. We should just clean up the bucket every so often to make sure this doesn't explode, I guess?